language/move-prover/bytecode/src/function_target_pipeline.rs

```diff
@@ -82,8 +91,9 @@ pub trait FunctionTargetProcessor {
     fn process(
         &self,
         _targets: &mut FunctionTargetsHolder,
-        _fun_env: &FunctionEnv<'_>,
+        _fun_env: &FunctionEnv,
         _data: FunctionData,
+        _scc_opt: Option<&[FunctionEnv]>,
     ) -> FunctionData {
```

Comment:
This is the main change proposed in this PR.

The `scc_opt: Option<&[FunctionEnv]>` is an argument passed to the entrypoint of every bytecode processor to indicate whether the current analysis target (`_fun_env` and `_data`) belongs to part of a recursive function group.
- `None` means this function is not recursive and
- `Some(scc)` means this function is recursive and `scc` is the list of mutually recursive functions